cogeo_mosaic/backends/dynamodb.py

```diff
@@ -52,9 +52,35 @@ def write(self):
         items = self._create_items()
         self._write_items(items)
 
-    def update(self):
+    def _update_quadkey(self, quadkey: str, dataset: List[str]):
+        """Update quadkey list."""
+        self.table.put_item(Item={"quadkey": quadkey, "assets": dataset})
+
+    def _update_metadata(self, bounds: List[float], version: str):
+        """Update bounds and center."""
+        meta = json.loads(json.dumps(self.metadata), parse_float=Decimal)
+        meta["version"] = version
+        meta["bounds"] = bounds
+        meta["center"] = (
+            (bounds[0] + bounds[2]) / 2,
+            (bounds[1] + bounds[3]) / 2,
+            self.mosaic_def.minzoom,
+        )
+        meta["quadkey"] == "-1"
+        self.table.put_item(Item=meta)
+
+    def update(
+        self,
+        features: Sequence[Dict],
+        accessor: Callable,
+        overwrite: bool = False,
+        **kwargs: Any
+    ):
         """Update the mosaicjson document."""
-        raise NotImplementedError
+        if not overwrite:
+            raise Exception("Overwrite has to be set to True")
```

Comment:
I don't understand this. If anything, it _can't_ be `True`, because to me an "overwrite" is equivalent to "delete a file and write a new file". The only way for that to happen with DynamoDB is to delete the table and create it again, or delete every individual quadkey.

So essentially only an "update" or "append" is possible with dynamodb.

Reply:
> So essentially only an "update" or "append" is possible with dynamodb.

I think I had to add the overwrite option to be consistent with the other backend https://github.com/developmentseed/cogeo-mosaic/blob/updateMosaic/cogeo_mosaic/scripts/cli.py#L182

Reply:
As written it does overwrite https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/dynamodb.html#DynamoDB.Client.put_item

Reply:
Right but that overwrites the _quadkey_, not the _mosaic_. With other backends, overwriting always overwrites the entire _file_, so this behavior isn't consistent.

Reply:
> I don't understand this. If anything, it can't be True, because to me an "overwrite" is equivalent to "delete a file and write a new file".

yeah to me overwrite, mean editing a file in-place.

> With other backends, overwriting always overwrites the entire file, so this behavior isn't consistent.

Yeah if you have better phrasing that works for all the backend I'm totally open to change it!

Reply:
>  If anything, it can't be True

well I'm not sure to understand, for DynamoDB backend `overwrite` has to be set to True.

Reply:
Other backends use a file as a representation of the mosaic.  When the update method is called with `overwrite=True` it replace the file with a new one containing the updated metadata and quadkeys.  Dynamo is different representation but same behavior ([overwrite metadata](https://github.com/developmentseed/cogeo-mosaic/blob/updateMosaic/cogeo_mosaic/backends/dynamodb.py#L70) and [overwrite quadkeys](https://github.com/developmentseed/cogeo-mosaic/blob/updateMosaic/cogeo_mosaic/backends/dynamodb.py#L57)).

Reply:
> yeah to me overwrite, mean editing a file in-place.

I think we have differing definitions. With a normal file, there's no such think as an "update in place", because you read the entire file and overwrite _the entire file_. It's not so much an update as it is deleting everything and writing new content.

Since DynamoDB is a database, my point is that it doesn't modify quadkeys that it doesn't read, so I think `overwrite` is a bit of a misnomer.

Reply:
Even using what I think is your definition of `overwrite`, when `overwrite` is `False`, it updates the mosaic locally but not remotely. I don't see a reason why that isn't possible for DynamoDB?

Reply:
and my question was do you think of a word that will fit for all the backend ?

Reply:
I need to think about it a bit more. At first glance, I would say that the `update` method _on the backend class_ should always commit the updates.

Reply:
yeah I think this comes down to, should we or should we not write the update! we will endup not having an update method for http backends but I'm fine with this

Reply:
> yeah I think this comes down to, `should we or should we not write the update!`If we choose to alway `write` we will end up not having an update method for http backends but I'm fine with this

@geospatial-jeff @kylebarron let me know what you think!